request-handler/src/main/java/au/org/aodn/aws/wps/operation/ExecuteOperation.java

```diff
@@ -102,8 +106,42 @@ public String execute() throws OGCException {
             s3JobFileManager.write(statusDocument, statusFileName, STATUS_FILE_MIME_TYPE);
 
             if (email != null) {
+                ExecuteRequestHelper requestHelper = new ExecuteRequestHelper(executeRequest);
+                String layer = requestHelper.getLiteralInputValue(LITERAL_INPUT_IDENTIFIER_LAYER);
+                String subset = requestHelper.getLiteralInputValue(LITERAL_INPUT_IDENTIFIER_SUBSET);
+                SubsetParameters subsetParams = SubsetParameters.parse(subset);
+
+                String collectionTitle = layer;
+
+                // Lookup collection metadata to get the collection title
+                try {
+                    //  Read the metadata record for the layer
+                    String catalogueURL = WpsConfig.getProperty(GEONETWORK_CATALOGUE_URL_CONFIG_KEY);
```

Comment:
Looks similar to code in aggregationrunner - can we pull this out into common?

Reply:
Maybe once day  ;)  (I'll look at it)